src/js/mixins/SaveStateMixin.js

```diff
@@ -19,6 +19,10 @@ const SaveStateMixin = {
   },
 
   componentWillUnmount() {
+    this.saveState();
+  },
+
+  saveState() {
```

Comment:
nice. good choice on the name. `saveState_saveState` would've been kind of repetitive